src/vinyldns/batch_change.py

```diff
@@ -51,16 +51,18 @@ def from_dict(d):
 
 
 class DeleteRecordSet(object):
-    def __init__(self, input_name, type):
+    def __init__(self, input_name, type, record=None):
         self.input_name = input_name
         self.type = type
+        self.record = record
         self.change_type = 'DeleteRecordSet'
 
     @staticmethod
     def from_dict(d):
         return DeleteRecordSet(
             input_name=d['inputName'],
-            type=d['type']
+            type=d['type'],
+            record=d.get('record')
```

Comment:
this will just get the `dict`, you need to convert it to a proper record.  The following snippet is pulled from `RecordSet`

`records=[rdata_converters[d['type']](rd) for rd in d.get('records', [])],`